### PR TITLE
Split sync of ApiService to sync and save

### PR DESCRIPTION
**Pull request by blizzz**
### 📝 Summary

As discussed with @juliushaertl when talking about #3899 it might be worthwhile to split the sync route into dedicated sync and save endpoints.

### 🚧 TODO

- [x] probably have to adjust cypress
- [x] take care of PublicSessionController

### 🏁 Checklist

- [ ] Code is properly formatted (`npm run lint` / `npm run stylelint` / `composer run cs:check`)
- [ ] [Sign-off message](https://probot.github.io/apps/dco/) is added to all commits
- [ ] [Tests](https://github.com/nextcloud/text#-testing-the-app) (unit, integration and/or end-to-end) passing and the changes are covered with tests
- [ ] Documentation ([README](https://github.com/nextcloud/text/blob/main/README.md) or [documentation](https://github.com/nextcloud/documentation/blob/master/admin_manual/configuration_server/text_configuration.rst#L2)) has been updated or is not required

### CODE REVIEW

File: src/services/SyncService.js [hunk 1]
```diff
@@ -242,7 +242,7 @@ class SyncService {
 	async save({ force = false, manualSave = true } = {}) {
```
Comment (juliushaertl) [inline src/services/SyncService.js:242]: We may still want to trigger the autosave somehow regular. I think this is no longer the case as only sync is called in an interval. Might be something to check with @max-nextcloud next week on how to best achieve that.

Comment (blizzz) [reply src/services/SyncService.js:242]: It's actually still being triggered regularly. At least yesterday it was :see_no_evil:

Comment (blizzz) [reply src/services/SyncService.js:242]: And that's the 30s autosave: https://github.com/nextcloud/text/blob/0d088d669fad3861387b27ca4e1b0cc51c147cab/src/services/SyncService.js#L88

**Comment by blizzz** [comment]
Pushed 2ac9c6c9ccc81a8a5289f1ef0a589e2421074d7b and 8f74dec8f but they don't appear here :shrug: :octocat::zzz:

File: lib/Service/ApiService.php [hunk 2]
```diff
@@ -233,14 +252,14 @@ public function sync(Session $session, Document $document, $version = 0, $autosa
 		}
 
 		try {
-			$result['document'] = $this->documentService->autosave($document, $file, $version, $autosaveContent, $documentState, $force, $manualSave, $token, $this->request->getParam('filePath'));
-		} catch (DocumentSaveConflictException $e) {
+			$result['document'] = $this->documentService->autosave($document, $file, $version, $autosaveContent, $documentState, $force, $manualSave, null, $this->request->getParam('filePath'));
```
Comment (max-nextcloud) [inline lib/Service/ApiService.php:255]: I think you'd still want to hand over the `token` - now called `shareToken` as provided as an argument.

```suggestion
			$result['document'] = $this->documentService->autosave($document, $file, $version, $autosaveContent, $documentState, $force, $manualSave, $shareToken, $this->request->getParam('filePath'));
```

**Review by max-nextcloud — CHANGES_REQUESTED** [review]
Thanks a lot for tackling this!
I have only one small remark.
Just went over the code did not test locally or go through untouched code to see if further changes would be needed.

File: lib/Service/ApiService.php (hunk 2 above)
Comment (blizzz) [reply lib/Service/ApiService.php:255]: Absolutely, good catch

Comment (max-nextcloud) [reply lib/Service/ApiService.php]: I applied the suggestion as a fixup to see if that helps with cypress.

**Review by max-nextcloud — APPROVED** [review]

**Comment by blizzz** [comment]
squahsed and rebassed

**Comment by blizzz** [comment]
The remaining cypress errors about conflicts are rooted that this information was not transported on `/sync` anymore. This is fixed now. It works locally, but Cypress is still not convinced.… will continue there.

The handler we have, reacts to this status only on polling! Saves would be ignored, you could see the failures only on the console (before and after the changes). With the polling itself it's not a big deal.

**Comment by blizzz** [comment]
The remaining issue with cypress conflicts tests was an accidentally inverted condition, so that the proper exception was not thrown.

**Pull request merged**